components/Scalar.php

```diff
@@ -0,0 +1,17 @@
+<?php
+
+namespace Components;
+
+/**
+ * Interface for scalar values.
+ */
+interface Scalar
+{
+
+    /**
+     * Returns calculated value of this scalar.
+     *
+     * @return mixed
+     */
+    public function value(): mixed;
+}
```

Comment:
This interface is made for objects which can be represented as a single value without any parameters.